huntsman/camera/pyro.py

```diff
@@ -575,20 +570,27 @@ def get_uid(self):
         """
         return self._camera.uid
 
-    def take_exposure(self, seconds, base_name, dark, *args, **kwargs):
-        # Using the /./ syntax for partial relative paths (needs rsync >= 2.6.7)
-        filename = os.path.join(os.path.abspath(self.config['directories']['images']),
-                                './',
-                                base_name)
+    def take_exposure(self, seconds, filename, dark, *args, **kwargs):
+        
+        #Make sure the filename is accessible on the SSHFS
+        basename = os.path.normpath(self.config['directories']['base'])
+        filename = os.path.normpath(filename)
+        if not filename.startswith(basename):
+            raise ValueError("Filename not accesible on SSHFS.")
```

Comment:
This is not really testing if it is accessible or not. Either the test or the message should change. `if os.path.exists(filename)`?

Reply:
wouldn't os.path.exists(filename) always return False anyway because the file hasn't been written yet? Perhaps something more like if os.path.isdir(os.path.dirname(filename))... ?